src/classes/presentation_controller.vala

```diff
@@ -1315,6 +1315,9 @@ namespace pdfpc {
 #if REST
             add_action("showQRcode", this.show_qrcode,
                 "Show QR code");
+#else
+            add_action("showQRcode", () => { stdout.printf("pdfpc not built with QR code support\n"); },
```

Comment:
IMO, no info/warning/error message should be printed. It will only annoy the user.

Reply:
> IMO, no info/warning/error message should be printed. It will only annoy the user.

Hmmm, noted. However, the message only gets printed out when the user actually calls the `showQRcode` function, it's not being printed every time the program is run. 

It was the only way I could think of to warn the user of the program's "build status". For instance, I wasn't aware of `pdfpc` QR code capabilities. 

I have no problem in removing the print statement if you feel like it would worsen the experience. :-D

Cheers!

Reply:
> However, the message only gets printed out when the user actually calls the showQRcode function, it's not being printed every time the program is run.

Yes, you're right. Yet, what's the purpose of this message? Maybe "annoy" isn't the right word. The real reason this action does nothing is not that the QR functionality wasn't compiled in but rather that there is no support for REST - and, hence, QR is not needed. So, if you can come up with a reasonably compact and meaningful text, that would be fine with me.

BTW, my initial suggestion was to modify `show_qrcode()` itself (with #if REST/#else) instead of introducing an anonymous closure. Finally, if a message is printed, it should use `GLib.printerr()` as the other warning messages do.

Reply:
> Yes, you're right. Yet, what's the purpose of this message? Maybe "annoy" isn't the right word. The real reason this action does nothing is not that the QR functionality wasn't compiled in but rather that there is no support for REST - and, hence, QR is not needed. So, if you can come up with a reasonably compact and meaningful text, that would be fine with me.

Perfect! I changed the message in the latest commit. I tried being as brief and meaningful as possible (or at least to the best of my abilities :) ).  What do you think of the new message?

> BTW, my initial suggestion was to modify `show_qrcode()` itself (with #if REST/#else) instead of introducing an anonymous closure. Finally, if a message is printed, it should use `GLib.printerr()` as the other warning messages do.

Noted, I changed it in the latest commit. I decided to re implement the functions twice simply because that's the Linux kernel guidelines. If you prefer a different approach (such us using the macro inside the function), I'll change it.

Cheers!